### PR TITLE
Migrate to react-admin v4

### PR DESCRIPTION
**Pull request by djhi**
First step for version 2, migrating to react-admin v4.

This PR focuses on making the main features work with RA4:
- data fetching
- authentication

Not tested:
- initial password page after invite. Will work on that in a dedicated PR before releasing v2

## Tasks

- [x] migrate dataProvider to ra-data-postgrest: https://github.com/marmelab/ra-supabase/pull/24
- [x] Add e2e tests
- [x] replicate supabase [Auth UI](https://supabase.com/docs/guides/auth/auth-helpers/auth-ui) with MUI (https://github.com/marmelab/ra-supabase/pull/26)
- [ ] document

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,9 +52,27 @@ prettier: ## prettify the source code using prettier
 test: build test-unit lint ## launch all tests
 
 test-unit: ## launch unit tests
-	echo "Running unit tests...";
-	yarn -s test-unit;
+	@echo "Running unit tests...";
+	@yarn -s test-unit;
```
Comment (slax57) [inline Makefile:56]: FMI, what's the purpose of the '@' symbol?

File: Makefile [hunk 2]
```diff
@@ -52,9 +52,27 @@ prettier: ## prettify the source code using prettier
...
+	@echo "Running unit tests..."; 
+	@yarn -s test-unit --watch;
+
+supabase-start: ## start the supabase server
+	@echo "Starting supabase server..."
+	@yarn supabase start
+
+supabase-Stop: ## stop the supabase server
+	@echo "Stoping supabase server..."
+	@yarn supabase Stop
```
Comment (slax57) [inline Makefile:68]:
```suggestion
supabase-stop: ## stop the supabase server
	@echo "Stopping supabase server..."
	@yarn supabase stop
```

File: packages/demo/README.md [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
-This version uses [ra-supabase](https://github.com/marmelab/ra-supabase).
+This is a demo of the [react-admin](https://github.com/marmelab/react-admin) library for React.js. It's a CRM for a fake Web agency with a few sales. You can test it online at https://marmelab.com/react-admin-crm.
+
+https://user-images.githubusercontent.com/99944/116970434-4a926480-acb8-11eb-8ce2-0602c680e45e.mp4
+
+React-admin usually requires a REST/GraphQL server to provide data. In this demo however, the API is simulated by the browser (using [FakeRest](https://github.com/marmelab/FakeRest)). The source data is generated at runtime by a package called [data-generator](https://github.com/marmelab/react-admin/tree/master/examples/data-generator).
+
+To explore the source code, start with [src/App.tsx](https://github.com/marmelab/react-admin/blob/master/examples/crm/src/App.tsx).
+
+**Note**: This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).
```
Comment (slax57) [inline packages/demo/README.md:11]: But now it seems to use vite, doesn't it?

File: packages/ra-supabase-core/src/dataProvider.ts [hunk 4]
```diff
@@ -1,34 +1,42 @@
-import { DataProvider, GetListParams } from 'ra-core';
+import { DataProvider, GetListParams, RaRecord } from 'ra-core';
```
Comment (slax57) [inline packages/ra-supabase-core/src/dataProvider.ts:1]: linter warning

File: packages/ra-supabase-core/src/useRedirectIfAuthenticated.ts [hunk 5]
```diff
@@ -15,19 +15,19 @@ import { useHistory } from 'react-router';
...
             .then(() => {
                 // already authenticated, redirect to the home page
-                history.push(redirectTo);
+                navigate(redirectTo);
             })
             .catch(() => {
                 // not authenticated, stay on the login page
             });
-    }, [checkAuth, history, redirectTo]);
+    }, [redirectTo]);
```
Comment (slax57) [inline packages/ra-supabase-core/src/useRedirectIfAuthenticated.ts:30]: linter warning

File: packages/ra-supabase-ui-materialui/src/AuthLayout.tsx [hunk 6]
```diff
@@ -8,15 +8,18 @@ import React, {
...
-import { createTheme, makeStyles } from '@material-ui/core/styles';
-import { ThemeProvider } from '@material-ui/styles';
-import LockIcon from '@material-ui/icons/Lock';
-import { StaticContext } from 'react-router';
+import {
+    Card,
+    Avatar,
+    ThemeProvider,
+    createTheme,
+    Box,
```
Comment (slax57) [inline packages/ra-supabase-ui-materialui/src/AuthLayout.tsx:16]: linter warning

File: packages/ra-supabase-ui-materialui/src/AuthLayout.tsx [hunk 7]
```diff
@@ -8,15 +8,18 @@ import React, {
...
+    styled,
+} from '@mui/material';
+import LockIcon from '@mui/icons-material/Lock';
 
 import { TitleComponent } from 'ra-core';
-import { defaultTheme, Notification } from 'ra-ui-materialui';
+import { defaultTheme, LayoutClasses, Notification } from 'ra-ui-materialui';
```
Comment (slax57) [inline packages/ra-supabase-ui-materialui/src/AuthLayout.tsx:22]: linter warning

File: Makefile (hunk 1 above)
Comment (djhi) [reply Makefile:56]: It ensure make does not print the command itself but just its result

File: packages/demo/README.md (hunk 3 above)
Comment (djhi) [inline packages/demo/README.md:11]:
```suggestion
**Note**: This project was bootstrapped with [Vite](https://vitejs.dev/).
```

**Review by slax57 — APPROVED** [review]

**Pull request merged**